sharktank/tests/kernels/wave/extend_attention_test.py

```diff
@@ -261,9 +295,167 @@ def test_no_cache(
 
         v_noise = v * 0.05
         extend_attention_v_noise = ops.extend_attention(
-            q=q, k=k, v=v_noise, start_positions=start_positions, seq_lens=seq_lens
+            q=q,
+            k=k,
+            v=v_noise,
+            kv_cache=None,
+            k_indices=indices_no_cache,
+            v_indices=indices_no_cache,
+            start_positions=start_positions,
+            seq_lens=seq_lens,
         )
         with pytest.raises(AssertionError):
             torch.testing.assert_close(
                 sdpa, extend_attention_v_noise, atol=1e-3, rtol=1e-3
             )
+
+    @pytest.mark.skipif(not torch.cuda.is_available(), reason="Needs CUDA/HIP device.")
+    @pytest.mark.parametrize(
+        "batch, heads, seq_len, head_dim, attn_dtype, device",
+        [
+            (1, 4, 32, 16, torch.float16, "cuda"),
+        ],
+    )
+    def test_extend_kv_cache_single_request_two_chunks(
+        self,
+        batch,
+        heads,
+        seq_len,
+        head_dim,
+        attn_dtype,
+        device,
+    ):
+        """Test extend attention over two sequential chunks for a single request."""
+        torch.manual_seed(42)
+
+        transformer_block_count = 8
+        transformer_block_index = 3
+        block_seq_stride = 4
+        chunk_size = 16  # tokens per chunk
+        num_chunks = seq_len // chunk_size
+
+        # Full QKV for reference
+        q_full = torch.randn(
+            batch, seq_len, heads, head_dim, dtype=attn_dtype, device=device
+        )
+        k_full = torch.randn(
+            batch, seq_len, heads, head_dim, dtype=attn_dtype, device=device
+        )
+        v_full = torch.randn(
+            batch, seq_len, heads, head_dim, dtype=attn_dtype, device=device
+        )
+
+        # Full SDPA reference (no KV cache)
+        q_sdpa = q_full.transpose(1, 2)
+        k_sdpa = k_full.transpose(1, 2)
+        v_sdpa = v_full.transpose(1, 2)
+        input_mask = ops.input_mask(torch.tensor([seq_len]), seq_len)
+        a = ops.attention_mask(
+            input_mask,
+            source_len=seq_len,
+            target_len=seq_len,
+            attention_dtype=attn_dtype,
+        ).to(device)
+        sdpa_ref = ops.scaled_dot_product_attention(q=q_sdpa, k=k_sdpa, v=v_sdpa, a=a)
+
+        # Build KV cache
+        page_count = batch * seq_len // block_seq_stride
+        kv_cache_extend = build_cache(
+            transformer_block_count=transformer_block_count,
+            attn_head_count=heads,
+            attn_head_dim=head_dim,
+            block_seq_stride=block_seq_stride,
+            cache_dtype=attn_dtype,
+            extend_attention=True,
+        )
+
+        cache = PagedGQAttention(
```

Comment:
This is inverted - the subcomponent will be invoking the kernel so they test should be using it externally.